django/curator/tag_deduplication.py

```diff
@@ -0,0 +1,162 @@
+import abc
+import os
+import re
+from typing import List
+
+import dedupe
+from django.utils.text import slugify
+from taggit.models import Tag
+
+from curator.models import CanonicalTagMapping, CanonicalTag
+
+
+class TagPreprocessing:
```

Comment:
in general class names are nouns, not verbs. So this would be better named `TagPreprocessor`

Similar things apply to all the other class names

`AbstractTagDeduper` or `BaseTagDeduper`

`TagClusterer`

`TagGazetteer`

Reply:
This seems unused as well, should it be?

Reply:
TagPreprocessor is supposed to be used as an upstream task. It is currently not used yet, I must have accidentally included it.

Reply:
Ah ok, if its meant to be used then it should be fine to include it